packages/core/lib/middleware/mw_utils.d.ts

```diff
@@ -0,0 +1,56 @@
+import * as http from 'http';
+import Segment = require('../segments/segment');
+import IncomingRequestData = require('./incoming_request_data');
+
+export const defaultName: string | undefined;
+
+export const dynamicNaming: boolean;
+
+export const hostPattern: string | null;
+
+export function enableDynamicNaming(hostPattern?: string): void;
+
+export function processHeaders(req?: Partial<Pick<http.IncomingMessage, 'headers'>>): { [key: string]: string };
```

Comment:
why `http.IncomingMessage` and not `http.IncomingMessage | https.IncomingMessage`?